src/source/mod.rs

```diff
@@ -314,6 +360,68 @@ fn extract_zip(archive: &Path, target_directory: &Path) -> Result<(), SourceErro
         .map_err(|e| SourceError::ZipExtractionError(e.to_string()))?;
 
     move_extracted_dir(tmp_extraction_dir.path(), target_directory)?;
+    progress_bar.finish_with_message("Extracted...");
 
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use std::{fs::File, io::Write};
+
+    use crate::source::SourceError;
+
+    use super::extract_zip;
+
+    #[test]
+    fn test_extract_zip() {
+        // zip contains text.txt with "Hello, World" text
+        const HELLOW_ZIP_FILE: &[u8] = &[
+            80, 75, 3, 4, 10, 0, 0, 0, 0, 0, 244, 123, 36, 88, 144, 58, 246, 64, 13, 0, 0, 0, 13,
+            0, 0, 0, 8, 0, 28, 0, 116, 101, 120, 116, 46, 116, 120, 116, 85, 84, 9, 0, 3, 4, 130,
+            150, 101, 6, 130, 150, 101, 117, 120, 11, 0, 1, 4, 245, 1, 0, 0, 4, 20, 0, 0, 0, 72,
+            101, 108, 108, 111, 44, 32, 87, 111, 114, 108, 100, 10, 80, 75, 1, 2, 30, 3, 10, 0, 0,
+            0, 0, 0, 244, 123, 36, 88, 144, 58, 246, 64, 13, 0, 0, 0, 13, 0, 0, 0, 8, 0, 24, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 164, 129, 0, 0, 0, 0, 116, 101, 120, 116, 46, 116, 120, 116, 85,
+            84, 5, 0, 3, 4, 130, 150, 101, 117, 120, 11, 0, 1, 4, 245, 1, 0, 0, 4, 20, 0, 0, 0, 80,
+            75, 5, 6, 0, 0, 0, 0, 1, 0, 1, 0, 78, 0, 0, 0, 79, 0, 0, 0, 0, 0,
+        ];
+        let term = indicatif::InMemoryTerm::new(100, 100);
+        let multi_progress = indicatif::MultiProgress::new();
+        multi_progress.set_draw_target(indicatif::ProgressDrawTarget::term_like(Box::new(
+            term.clone(),
+        )));
+        let tempdir = tempfile::tempdir().unwrap();
+        let file_path = tempdir.path().join("test.zip");
+        let mut file = File::create(&file_path).unwrap();
+        _ = file.write_all(HELLOW_ZIP_FILE);
+
+        let res = extract_zip(file_path, tempdir.path(), multi_progress.clone());
+        assert!(term.contents().trim().starts_with(
+            "Extracting zip       [00:00:00] [━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━]"
+        ));
+        assert!(matches!(res.err(), None));
```

Comment:
```suggestion
        assert!(res.err().is_none());
```

Reply:
you can also just `unwrap` in the test and it should fail the test if there is an error.